src/github.com/getlantern/peerscanner/cf/cf.go

```diff
@@ -0,0 +1,82 @@
+// package cf provides a utility for interacting with CloudFlare
+package cf
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/getlantern/cloudflare"
+	"github.com/getlantern/golog"
+)
+
+var (
+	log = golog.LoggerFor("cf")
+)
+
+type Util struct {
+	Client *cloudflare.Client
+	domain string
+}
+
+func New(domain string, username string, apiKey string) *Util {
+	client := cloudflare.NewClient(username, apiKey)
+	// Set a longish timeout on the HTTP client just in case
+	client.Http = &http.Client{
+		Timeout: 5 * time.Minute,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				ClientSessionCache: tls.NewLRUClientSessionCache(1000),
+			},
+		},
+	}
+	return &Util{client, domain}
+}
+
+func (util *Util) GetAllRecords() ([]cloudflare.Record, error) {
+	resp, err := util.Client.LoadAll(util.domain)
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving records!", err)
+	}
+
+	allRecords := resp.Response.Recs.Records
+	for resp.Response.Recs.HasMore {
+		resp, err = util.Client.LoadAllAtIndex(util.domain, len(allRecords))
+		if err != nil {
+			return nil, fmt.Errorf("Error retrieving records at index!", err)
```

Comment:
`fmt.Errorf("Error retrieving records at index!", err)` does not actually include the format code for displaying `err`, so it will get appended as `%!(EXTRA ...)`. We could fix it with `fmt.Errorf("Error retrieving records at index!: %q", err)`.